vault.go

```diff
@@ -94,22 +95,32 @@ func (v *Vault) login(ctx context.Context) (*vault.Secret, error) {
 }
 
 // GetSecretAPIKey fetches the latest version of secret api key from kv-v2
-func (v *Vault) GetSecretAPIKey(ctx context.Context) (map[string]interface{}, error) {
+func (v *Vault) GetSecretAPIKey(ctx context.Context) (string, error) {
 	log.Println("getting secret api key from vault")
 
 	secret, err := v.client.Logical().Read(v.parameters.apiKeyPath)
 	if err != nil {
-		return nil, fmt.Errorf("unable to read secret: %w", err)
+		return "", fmt.Errorf("unable to read secret: %w", err)
 	}
 
 	data, ok := secret.Data["data"].(map[string]interface{})
 	if !ok {
-		return nil, fmt.Errorf("malformed secret returned")
+		return "", fmt.Errorf("malformed secret returned")
+	}
+
+	apiKey, ok := data[v.parameters.apiKeyField]
+	if !ok {
+		return "", fmt.Errorf("the secret retrieved from vault is missing %q field", v.parameters.apiKeyField)
+	}
+
+	apiKeyString, ok := apiKey.(string)
```

Comment:
I think this is great for now. In the future, I think it would make sense to serialize and deserialize our own secret shape so there is a little more "meat" in writing and retrieving structured data.
e.g.
```go
type ApiSecret struct {
    api-key string
    api-user string
}
```

Reply:
Agreed